decred/tests/unit/dcr/test_txscript.py

```diff
@@ -3318,3 +3317,142 @@ def test_get_P2PKH_pCode():
+    idx (int): The output index that pkScript belongs to.
+    pkScript (ByteArray): The script that spends output idx of tx.
+    scriptClass (int): Type of script.
+    addresses list(str): Addresses that comprise a multisig.
+    nRequired (int): Number of addresses required for a multisig.
+    sigScript (ByteArray): The signed script.
+    prevScript (ByteArray): The previous script.
+    want (ByteArray): Merged script.
+    """
+    tests = [dict(
+        name="pay to public key hash",
+        pkScript=P2PKH,
+        scriptClass=txscript.PubKeyHashTy,
+        sigScript=P2PKHsig,
+        want=P2PKHsig,
+    ), dict(
+        name="pay to public key hash prevScript longer",
+        pkScript=P2PKH,
+        scriptClass=txscript.PubKeyHashTy,
+        sigScript=P2PKHsig,
+        prevScript=P2PKHsig + emptySig1x,
+        want=P2PKHsig + emptySig1x,
+    ), dict(
+        name="3 of 4 multisig one signature",
+        tx=tx,
+        pkScript=multisig,
+        addresses=[pub1],
+        nRequired=3,
+        scriptClass=txscript.MultiSigTy,
+        sigScript=multisigSig1 + emptySig2x,
+        want=multisigSig1 + emptySig2x,
+    ), dict(
+        name="3 of 4 multisig two signatures",
+        tx=tx,
+        pkScript=multisig,
+        addresses=[pub1, pub2],
+        nRequired=3,
+        scriptClass=txscript.MultiSigTy,
+        sigScript=multisigSig1 + multisigSig2 + emptySig1x,
+        want=multisigSig1 + multisigSig2 + emptySig1x,
+    ), dict(
+        name="3 of 4 multisig merging two signatures",
+        tx=tx,
+        idx=0,
+        pkScript=multisig,
+        addresses=[pub1, pub2],
+        nRequired=3,
+        scriptClass=txscript.MultiSigTy,
+        sigScript=multisigSig2 + emptySig2x,
+        prevScript=multisigSig1 + emptySig2x,
+        want=multisigSig1 + multisigSig2 + emptySig1x,
+    ), dict(
+        name="3 of 4 multisig pay to script hash",
+        tx=tx,
+        idx=0,
+        pkScript=multisigP2SH,
+        addresses=[pub1],
+        nRequired=3,
+        scriptClass=txscript.ScriptHashTy,
+        sigScript=multisigSig1 + emptySig2x + txscript.addData(multisig),
+        want=multisigSig1 + emptySig2x + txscript.addData(multisig),
+    ), dict(
+        name="3 of 4 multisig pay to script hash merging three signatures",
+        tx=tx,
+        idx=0,
+        pkScript=multisigP2SH,
+        addresses=[pub3, pub2, pub1],
+        nRequired=3,
+        scriptClass=txscript.ScriptHashTy,
+        sigScript=multisigSig2 + emptySig2x + txscript.addData(multisig),
+        prevScript=multisigSig1 + multisigSig3 + emptySig1x + txscript.addData(multisig),
+        want=multisigSig1 + multisigSig2 + multisigSig3 + txscript.addData(multisig),
+    )]
+
+    for test in tests:
+        res = txscript.mergeScripts(testnet, test.get("tx"), test.get("idx"), test.get("pkScript"), test.get("scriptClass"), test.get("addresses"), test.get("nRequired"), test.get("sigScript"), test.get("prevScript"))
```

Comment:
It's surprising that Black is allowing these long lines. We might want `flake8` to be stricter about line length, with local exceptions of course.

Reply:
Actually, it looks like there are more fmt off than fmt on, so nothing in the entire file is being formatted. My fault.